boot.go

```diff
@@ -82,8 +82,12 @@ func main() {
 				log.Printf("Starting health check server on port %d", cnf.HealthSrvPort)
 				healthSrvCh := make(chan error)
 				go func() {
-					if err := healthsrv.Start(cnf.HealthSrvPort, kubeClient.Namespaces(), storageDriver, circ); err != nil {
-						healthSrvCh <- err
+					for{
+						if err := healthsrv.Start(cnf.HealthSrvPort, kubeClient.Namespaces(), storageDriver, circ); err != nil {
+							healthSrvCh <- err
+							break
+						}
+						log.Print("namespace watcher channel closed.Starting the watch again")
```

Comment:
@arschles i think i misunderstood your point....do you mean exiting of the builder process? i am not sure if there is need to restart the builder when the watcher closes its channel as it  doesn't effect any builder flow?